src/Compilers/Core/Private/Microsoft.CodeAnalysis.Private.csproj

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
Unfortunately, I needed to create new project which contains the custom `internal class ObsoleteAttribute`, because if it was included in `Microsoft.CodeAnalysis`, it would be visible to all other IVT projects, leading to ambiguity warnings (or even errors in Visual Basic).

Reply:
It would be good to document (somehow) what is the purpose of this project. What types can/should be added here?

Reply:
It's shortly documented in `<PackageDescription>` field in this file - it's for private utility types, i.e., types that should not be IVT to other assemblies unlike types in `Microsoft.CodeAnalysis`. I'm not sure if I should be that specific in `<PackageDescription>`, I guess that's visible on NuGet. But I can add a comment.

Reply:
If we did go this approach this is still missing a bit more of a "why?" here -- we needed this because we had to redefine a framework type and that caused downstream problems. But for example it'd be safe to remove if we retargeted back to something that defined the DiagnosticId on the ObsoleteAttribute.